### PR TITLE
Add prefix to org, space, process, and route GUIDS

### PR DESCRIPTION
**Pull request by clintyoshimura**
## Is there a related GitHub Issue?
#724 

## What is this change about?
Add prefixes to orgs, spaces, processes, and routes to allow operators to identify resources created by cf

## Does this PR introduce a breaking change?
No

## Acceptance Steps
See issue

## Tag your pair, your PM, and/or team
@julian-hj

### CODE REVIEW

File: api/apis/job_handler.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func (h *JobHandler) RegisterRoutes(router *mux.Router) {
 
 func parseJobGUID(jobGUID string) (string, string, bool) {
 	// Match job.type-GUID and capture the job type and GUID for later use
-	jobRegexp := regexp.MustCompile("([a-z_-]+[.][a-z_]+)-([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})")
+	jobRegexp := regexp.MustCompile("([a-z_-]+[.][a-z_]+)-([a-z]*-?[a-z]*-?[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})")
```
Comment (kieron-dev) [inline api/apis/job_handler.go:73]: This regex is becoming quite complex. I wonder if the new optional part for matching `cf-xxx-` could be made clearer by using a non-capturing group. E.g. `(?:cf-[a-z]+-)?` instead of `[a-z]*-?[a-z]*-?`. Or if there is a finite (for now) list of prefixes, it could be more explicit with `(?:cf-(?:org|space|route|proc)-)?` maybe?

File: api/apis/org_handler.go [hunk 2]
```diff
@@ -58,7 +59,7 @@ func (h *OrgHandler) orgCreateHandler(info authorization.Info, r *http.Request)
 	}
 
 	org := payload.ToMessage()
-	org.GUID = uuid.NewString()
+	org.GUID = OrgPrefix + uuid.NewString()
```
Comment (kieron-dev) [inline api/apis/org_handler.go:62]: I notice we are generating the GUID in 3 different layers: handler for orgs and spaces, repo for routes and the app reconciler for processes. Maybe this PR is a good opportunity to move the org and space GUID generation to the repo layer to make things more uniform. The process GUID will have to stay in the app reconciler.

**Review by kieron-dev — COMMENTED** [review]
This looks generally fine to me. Just see what you think about suggestions about the hairy regex and whether we should move the GUID generation for orgs and spaces.

File: api/tests/e2e/service_instances_test.go [hunk 3]
```diff
@@ -123,13 +123,15 @@ var _ = Describe("Service Instances", func() {
 			})
 
 			It("deletes the service instance", func() {
```
Comment (kieron-dev) [inline api/tests/e2e/service_instances_test.go:125]: This appears unrelated to the story. Maybe better in a separate PR?

File: api/apis/job_handler.go (hunk 1 above)
Comment (clintyoshimura) [reply api/apis/job_handler.go]: Updated to use the `(?:cf-[a-z]+-)?` format. It seems like that would be the best solution without having to add to the list of types if we add more.

File: api/apis/org_handler.go (hunk 2 above)
Comment (clintyoshimura) [reply api/apis/org_handler.go]: Moved the Org/space uuid generation to the repository layer.

File: api/tests/e2e/service_instances_test.go (hunk 3 above)
Comment (clintyoshimura) [reply api/tests/e2e/service_instances_test.go]: This was only because i couldn't get it to run locally. It doesn't seem to be an issue when git runs the e2e test so I reverted it.

**Review by matt-royal — APPROVED** [review]
We paired to wrap this up and address the feedback. LGTM
